netcdfcxx4: 4.3.0 -> 4.3.1

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/netcdf-cxx4/versions.



meta.description for netcdfcxx4 is: '"C++ API to manipulate netcdf files"'.


meta.homepage for netcdfcxx4 is: '"https://www.unidata.ucar.edu/software/netcdf/"

[Release on GitHub](https://github.com/Unidata/netcdf-cxx4/releases/tag/v4.3.1)


[Compare changes on GitHub](https://github.com/Unidata/netcdf-cxx4/compare/v4.3.0...v4.3.1)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 4.3.1 with grep in /nix/store/fw2q16c302kmasi4gg6q807xicgnvv8k-netcdf-cxx4-4.3.1
- directory tree listing: https://gist.github.com/84fcda2d7f372d4e6640f4a85720bfd3

- du listing: https://gist.github.com/e3ec0e948193e42fb45d1168a296cee2


</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

7 total rebuild path(s)

2 package rebuild(s)

2 x86_64-linux rebuild(s)
2 i686-linux rebuild(s)
1 x86_64-darwin rebuild(s)
2 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
nco
netcdfcxx4

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/fw2q16c302kmasi4gg6q807xicgnvv8k-netcdf-cxx4-4.3.1 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A netcdfcxx4 https://github.com/r-ryantm/nixpkgs/archive/b26c87acf46ec0b0e9f8c3e6050dea6acef76d24.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/fw2q16c302kmasi4gg6q807xicgnvv8k-netcdf-cxx4-4.3.1
ls -la /nix/store/fw2q16c302kmasi4gg6q807xicgnvv8k-netcdf-cxx4-4.3.1/bin
```


</details>
<br/>

r-ryantm committed:
netcdfcxx4: 4.3.0 -> 4.3.1

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/netcdf-cxx4/versions

c0bw3b committed:
netcdfcxx4: use fetchzip

Github archives are not stable over time so
fetchzip should be used instead of fetchurl

c0bw3b committed:
netcdfcxx4: fix linking on Darwin

c0bw3b commented:
Yup, now there is a second Darwin-only linking issue with `hdf5`

FRidh commented:
@c0bw3b this needed more work?

ryantm commented:
@GrahamcOfBorg eval

SuperSandro2000 commented:
LGTM. Can this be merged now?

ryantm commented:
@SuperSandro2000 does it work on darwin for you?

SuperSandro2000 commented:
@r-ryantm nope

```
Making all in plugins
make[2]: Entering directory '/private/tmp/nix-build-netcdf-cxx4-4.3.1.drv-0/source/plugins'
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o H5Zbzip2.lo H5Zbzip2.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c H5Zbzip2.c  -fno-common -DPIC -o .libs/H5Zbzip2.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o blocksort.lo blocksort.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c blocksort.c  -fno-common -DPIC -o .libs/blocksort.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o huffman.lo huffman.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c huffman.c  -fno-common -DPIC -o .libs/huffman.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o crctable.lo crctable.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c crctable.c  -fno-common -DPIC -o .libs/crctable.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o randtable.lo randtable.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c randtable.c  -fno-common -DPIC -o .libs/randtable.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o compress.lo compress.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c compress.c  -fno-common -DPIC -o .libs/compress.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o decompress.lo decompress.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c decompress.c  -fno-common -DPIC -o .libs/decompress.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I../cxx4   -g -O2 -c -o bzlib.lo bzlib.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I../cxx4 -g -O2 -c bzlib.c  -fno-common -DPIC -o .libs/bzlib.o
/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash ../libtool  --tag=CC   --mode=link clang  -g -O2 -module -avoid-version -shared -export-dynamic -no-undefined  -o libh5bzip2.la -rpath /nix/store/m21s2xx1rvyyj288jrg6d5hb0k1nw4xj-netcdf-cxx4-4.3.1/lib H5Zbzip2.lo blocksort.lo huffman.lo crctable.lo randtable.lo compress.lo decompress.lo bzlib.lo    -lnetcdf
libtool: link: clang  -o .libs/libh5bzip2.so -bundle  .libs/H5Zbzip2.o .libs/blocksort.o .libs/huffman.o .libs/crctable.o .libs/randtable.o .libs/compress.o .libs/decompress.o .libs/bzlib.o   -lnetcdf  -g -O2
Undefined symbols for architecture x86_64:
  "_H5free_memory", referenced from:
      _H5Z_filter_bzip2 in H5Zbzip2.o
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Makefile:468: libh5bzip2.la] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-netcdf-cxx4-4.3.1.drv-0/source/plugins'
make[1]: *** [Makefile:543: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-netcdf-cxx4-4.3.1.drv-0/source'
make: *** [Makefile:413: all] Error 2
```

ryantm commented:
Result of `nixpkgs-review pr 69692` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nco</li>
    <li>netcdfcxx4</li>
  </ul>
</details>

ryantm commented:
It is a regression on Darwin.
```
$ hydra-check  netcdfcxx4 --channel nixpkgs/nixpkgs-20.09-darwin --arch=x86_64-darwin
Build Status for netcdfcxx4.x86_64-darwin on nixpkgs/nixpkgs-20.09-darwin
✔ netcdf-cxx4-4.3.0 from 2020-09-20 - https://hydra.nixos.org/build/127602410
```

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 69692` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>netcdfcxx4: log <a href="https://termbin.com/41u1">https://termbin.com/41u1</a></li>
  </ul>
</details>

lukegb committed:
netcdfcxx4: switch to CMake

This resolves the Darwin linking issues Once and For All.

Also, this makes the build parallel, which is nice too.
